### ISSUE TITLE
User presence accounting problems seeing all people in a room 

### ISSUE THREAD

**Issue by jittygitty** [matrix-org/dendrite#2597]
### Background information

- **Git pull yesterday, so 0.8.9?**: 
- **Monolith**: 
- **SQLite3**: 
- **No Docker**: 
- **go version 1.17.7**:
- **Issue persists with any Client**:

### Description

 - **What** is the problem:
 - Users under settins/info "People" not showing all people, in element desktop showing self as offline, in android client those offline users not even listed at all, invisible
 - **Who** is affected:
 - Just about everyone using my Dendrite home server?
 - **How** is this bug manifesting:
 - "Manifesting" quite well for all my users
 - **When** did this first appear:
 - I think it first appeared from the "beginning" of my using matrix chat (i never used synapse)
 

### Steps to reproduce

- list the steps
- that reproduce the bug
- using hyphens as bullet points

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from server logs, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.

Alternatively, please send logs to @kegan:matrix.org or @neilalexander:matrix.org
with a link to the respective Github issue, thanks!
-->

**Comment by jittygitty** [comment]
Also, as I said in https://github.com/vector-im/element-web/issues/22955
It would be nice to have a workaround until this is fixed so that an Admin user can promote another user in channel even if they're invisible due to this bug, so be able to type in a username and promote its privileges etc. Can create a related feature request for this after feedback.

**Comment by neilalexander** [comment]
Please try with version 0.9.0 and let us know if it's still a problem.

**Comment by jittygitty** [comment]
@neilalexander I just upgraded to latest 0.9.3+bcdbd5c0 and when I click on people in settings in a room in Android client for example, I still only see myself inside there (even though it had said '2'). And on desktop client I see both of us in the room but it says "offline" when hovering over. Does it require that I do some "init sync" or something else other than restart dendrite?

**Comment by bekcpear** [comment]
I had the same problem, all of people, even me, showed up as 'offline' on element app.  Version 0.10.3

I want to debug it, but don't know how to start.

**Comment by jittygitty** [comment]
@bekcpear I've been too busy with other things, and so my Dendrite server I think is still on 0.9.3+bcdbd5c0 so can you check your Dendrite server version? Have you updated to the "latest" Dendrite 0.10.3 to see if it helps?

If you use Element "Desktop" app on windows, you can do a Ctrl-Shift-i  to open the debug/dev console etc.

**Comment by bekcpear** [comment]
@jittygitty 
> Have you updated to the "latest" Dendrite 0.10.3 to see if it helps?

Yes, my dendrite version is 0.10.3, and element version is `version: 1.11.10 Olm version: 3.2.12`. I confirm that everyone including myself is displayed as offline.

**Comment by bekcpear** [comment]
ummm, it's my fault, after set

```
presence:
    enable_inbound: true
    enable_outbound: true
```

The online status looks good now.

**Comment by S7evinK** [comment]
Right, without presence you won't see any online/offline users. Keep in mind that this can cause performance issues when in large/active rooms.
If you don't want to enable presence but hide the "offline" and greyed out users in Element Web, you'd need to host your own Element Web with a custom config, setting
```json
{
      "enable_presence_by_hs_url": {
        "https://yourHomserverDomain": false
      },
}
```
Does this resolve the issue?
